File: production/shared/src/main/scala/kpn/shared/changes/ChangeSetPage.scala
package kpn.shared.changes

import kpn.shared.ChangeSetSummary
import kpn.shared.changes.details.NetworkChangeInfo
import kpn.shared.common.KnownElements
import kpn.shared.node.NodeChangeInfo
import kpn.shared.route.RouteChangeInfo

case class ChangeSetPage(
  summary: ChangeSetSummary,
  changeSetInfo: Option[ChangeSetInfo],
  networkChanges: Seq[NetworkChangeInfo],
  routeChanges: Seq[RouteChangeInfo],
  nodeChanges: Seq[NodeChangeInfo],
  knownElements: KnownElements,
  reviews: Seq[Review]
)

File: backend/api/src/main/java/kpn/api/common/changes/ChangeSetPage.java
package kpn.api.common.changes;

import kpn.api.common.changes.ChangeSetDetail;

import com.google.common.collect.ImmutableList;

public record ChangeSetPage(
  ImmutableList<ChangeSetDetail> details
) {
}

/*
package kpn.api.common.changes

case class ChangeSetPage(
  details: Seq[ChangeSetDetail]
)

*/
